src/L2DebtManager.sol

```diff
@@ -1,262 +1,502 @@
+            address(weETH),
+            collaterals[0].amount
         );
 
-        SafeERC20.safeTransfer(USDC, etherFiCashSafe, borrowUsdcAmount);
+        return (collaterals, totalCollateralInUsd);
+    }
 
-        emit Borrowed(msg.sender, address(USDC), borrowUsdcAmount);
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function liquidatable(address user) public view returns (bool) {
+        return debtRatioOf(user) > _liquidationThreshold;
     }
 
-    /// Users repay the borrowed USDC in USDC
-    function repayWithUSDC(uint256 repayUsdcAmount) external {
-        _borrowings[msg.sender] -= repayUsdcAmount;
-        totalBorrowingAmount -= repayUsdcAmount;
-
-        SafeERC20.safeTransferFrom(
-            USDC,
-            msg.sender,
-            address(this),
-            repayUsdcAmount
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function collateralOf(
+        address user
+    ) public view returns (Collateral[] memory, uint256) {
+        Collateral[] memory collaterals = new Collateral[](1);
+        collaterals[0] = Collateral({
+            token: address(weETH),
+            amount: _userCollateral[user][address(weETH)]
+        });
+
+        uint256 totalCollateralInUsd = convertCollateralTokenToUsdc(
+            address(weETH),
+            collaterals[0].amount
         );
 
-        emit RepaidWithUSDC(msg.sender, repayUsdcAmount);
+        return (collaterals, totalCollateralInUsd);
     }
 
-    /// Users repay the borrowed USDC in eETH
-    /// Equivalent to partially liquidating the debt using the collateral
-    function repayWithEETH(uint256 repayUsdcAmount) external {
-        _repayWithEEthCollateral(msg.sender, repayUsdcAmount);
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function borrowingOf(address user) public view returns (uint256) {
+        return _userBorrowings[user];
     }
 
-    // https://docs.aave.com/faq/liquidations
-    /// Liquidate the user's debt by repaying the entire debt using the collateral
-    /// @dev do we need to add penalty?
-    function liquidate(address user) external {
-        require(liquidatable(user), "NO_LIQUIDATION_REQUIRED");
-
-        uint256 beforeDebtAmount = _borrowings[user];
-        uint256 beforeCollateralAmount = _collaterals[user];
-        _repayWithEEthCollateral(user, beforeDebtAmount); // force to repay the entire debt using the collateral
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function debtRatioOf(address user) public view returns (uint256) {
+        uint256 debtValue = _userBorrowings[user];
+        uint256 collateralValue = getCollateralValueInUsdc(user);
+        if (collateralValue == 0) revert ZeroCollateralValue();
 
-        emit Liquidated(
-            user,
-            beforeCollateralAmount,
-            _collaterals[user],
-            beforeDebtAmount
-        );
+        return (debtValue * 1e20) / collateralValue; // result in basis points
```

Comment:
how does the math work here? why 1e20?

Reply:
I was using the bps before for liquidation threshold but because the test was failing if I added 1 wei more than the expected limit of 60%. So I checked and it was an issue with the decimals. So I increased the precision by adding a total of 18 decimals after the percentage value in threshold. Hence 1e20.

Reply:
i see :) thx